src/Conn.cc

```diff
@@ -234,21 +319,30 @@ const RecordValPtr& Connection::GetVal() {
 
         if ( inner_vlan != 0 )
             conn_val->Assign(10, inner_vlan);
+
+        // This is the first time the conn_val is created.
+        // Let all the analyzers in the tree starting at
+        // the session adapter know about it, once.
+        //
+        // AddChildAnalyzer() will do it for new child
+        // analyzers added when conn_val exists.
+        //
+        // If the adapter isn't set yet, SetSessionAdapter()
+        // will do the job, too.
+        if ( adapter ) {
+            analyzer::for_each(*adapter, [this](analyzer::Analyzer& a) {
+                // Call InitConnVal() on all analyzer in this connection's analyzer tree.
```

Comment:
all analyzer -> all analyzers
OTOH, this comment seems obvious so you could just remove it